File: samples/c/08objects/rectangle.c
#include <stdlib.h>
#include "rectangle.h"


Rectangle* createRectangle(float width, float height) {
    Rectangle* this = malloc(sizeof(Rectangle));
    this->width = width;
    this->height = height;
}

float rectangleArea(Rectangle* r) {
    return r->width * r->height;
}

float rectanglePerimeter(Rectangle* r) {
    return r->width * 2 + r->height * 2;
}

float deleteRectangle(Rectangle** r) {
    free(*r);
    *r = NULL;
}

File: samples/java/08objects/main/Rectangle.java
package main;

import main.Shape;

public class Rectangle extends Shape {
    private float height;
    private float width;

    public Rectangle(float width, float height) {
        this.height = height;
        this.width = width;
    }

    public float perimeter() {
        return this.height * 2 + this.width * 2;
    }

    public float area() {
        return this.height * this.width;
    }
}
